grafana/grafana-operator
# Add environment variables configuring setting dashboard namespaces

## Description
Currently it's only possible to configure dashboard namespaces through
command line args. This makes it difficult to fine-tune the operator
when installing it declaratively through an OLM subscription.

This commit adds the `DASHBOARD_NAMESPACES` and `DASHBOARD_NAMESPACES_ALL`
environment variables which serve as fallback for the namespaces and
scan-all flags.

## Relevant issues/tickets
Fixes https://github.com/grafana-operator/grafana-operator/issues/363


## Type of change




## Checklist


## Verification steps
Run the operator with the following commands to verify the functionality:
* ```POD_NAME="grafana-operator" WATCH_NAMESPACE="default" go run main.go```  uses current namespace for dashboards
* ```POD_NAME="grafana-operator" WATCH_NAMESPACE="default" DASHBOARD_NAMESPACES_ALL="true" go run main.go```: uses all namespaces for dashboards
* ```POD_NAME="grafana-operator" WATCH_NAMESPACE="default" DASHBOARD_NAMESPACES="kube-system" go run main.go```: uses kube-system namespace for dashboards
*  ```POD_NAME="grafana-operator" WATCH_NAMESPACE="default" DASHBOARD_NAMESPACES="kube-system" go run main.go --namespaces default```: uses default namespace for dasbhoards
